### PR TITLE
fix(publick8s): add `nodeSelector` to migrated services

### PR DESCRIPTION
**Pull request by lemeurherve**
This PR ensures pods are assigned to [`x86medium` node pool](https://github.com/jenkins-infra/azure/blob/ecdcc7369199b8494f4b63fd28c240187ac61e49/publick8s.tf#L84-L96) in `publick8s`, and avoid mixing type of nodes for a same service.

~For smaller services, the cheaper [`arm64small` node pool](https://github.com/jenkins-infra/azure/blob/ecdcc7369199b8494f4b63fd28c240187ac61e49/publick8s.tf#L98-L110) has been privileged, while other has been assigned to the `publicpool` by default.~

Ref: https://github.com/jenkins-infra/helpdesk/issues/3351 ~& https://github.com/jenkins-infra/helpdesk/issues/3584~

### CODE REVIEW

File: config/artifact-caching-proxy_azure.yaml [hunk 1]
```diff
@@ -11,3 +11,6 @@ ingress:
 
 persistence:
   storageClass: managed-csi-premium-retain
+
+nodeSelector:
+  agentpool: arm64small
```
Comment (dduportal) [inline config/artifact-caching-proxy_azure.yaml:16]:
```suggestion

nodeSelector:
  agentpool: x86medium
```

Switching ACP to ARM looks risky for now, as the Nginx tuning is making assumption that might not be the same in ARM kernel + 4vCPUs on the underlying machine is not enough

File: config/incrementals-publisher.yaml [hunk 2]
```diff
@@ -12,3 +12,6 @@ ingress:
     - secretName: incrementals-tls
       hosts:
         - incrementals.jenkins.io
+
+nodeSelector:
+  agentpool: publicpool
```
Comment (dduportal) [inline config/incrementals-publisher.yaml:17]:
```suggestion
  agentpool: x86medium
```

Use the new node pool

File: config/jenkins_weekly.ci.jenkins.io.yaml [hunk 3]
```diff
@@ -21,6 +21,9 @@ controller:
   image: jenkinsciinfra/jenkins-weekly
   tag: 0.116.1-2.407
   pullPolicy: IfNotPresent
+  nodeSelector:
+    kubernetes.io/os: linux
+    kubernetes.azure.com/agentpool: arm64small
```
Comment (dduportal) [inline config/jenkins_weekly.ci.jenkins.io.yaml:26]:
```suggestion
    kubernetes.azure.com/agentpool: x86medium
```

We haven't evaluated the JVM behavior with Jenkins controller under ARM. Too risky

File: config/rating.yaml [hunk 4]
```diff
@@ -14,5 +14,8 @@ ingress:
       hosts:
         - rating.jenkins.io
 
+nodeSelector:
+  agentpool: arm64small
```
Comment (dduportal) [inline config/rating.yaml:18]:
```suggestion
  agentpool: x86medium
```

We haven't evaluated if an ARM image is available for this app

File: config/uplink.yaml [hunk 5]
```diff
@@ -13,9 +13,13 @@ ingress:
...
 image:
   # repo & '@sha256' suffix here as we're retrieving a docker digest from updatecli
   repository: jenkinsciinfra/uplink@sha256
   tag: c7ecad0e5e24bc05ac722418faf844d6a0286620c6316ff712459ecfcc3ee7ce
   pullPolicy: IfNotPresent
+
+replicaCount: 5
+
+nodeSelector:
+  agentpool: arm64small
```
Comment (dduportal) [inline config/uplink.yaml:25]:
```suggestion
  agentpool: x86medium
```

We haven't evaluated if an ARM image is available for this app

**Review by dduportal — CHANGES_REQUESTED** [review]
- Using the new `arm64` pool is under work by @smerle33 , I'm not sure it's wise to use it on all the services. Let's stick to x86
- We only want to "force" workloads to the `x86medium` node pool

File: config/jenkins_weekly.ci.jenkins.io.yaml (hunk 3 above)
Comment (lemeurherve) [reply config/jenkins_weekly.ci.jenkins.io.yaml:26]: It's a demo instance, wouldn't it be a perfect candidate to test it as no production impact if something goes wrong?

Could be in another PR

File: config/javadoc.yaml [hunk 6]
```diff
@@ -29,3 +29,8 @@ htmlVolume:
     secretName: javadoc
     shareName: javadoc
     readOnly: true
+
+replicaCount: 2
+
+nodeSelector:
+  agentpool: arm64small
```
Comment (lemeurherve) [inline config/javadoc.yaml:36]: @dduportal @smerle33 should I keep this node pool selection for `javadoc`? (cf https://github.com/jenkins-infra/helpdesk/issues/3584)

Or should I switch it to `x86medium`?
```suggestion
  agentpool: x86medium
```

File: config/jenkins_weekly.ci.jenkins.io.yaml (hunk 3 above)
Comment (dduportal) [reply config/jenkins_weekly.ci.jenkins.io.yaml]: It's way way way out of our scopes

Comment (timja) [reply config/jenkins_weekly.ci.jenkins.io.yaml]: definitely worth putting it on ARM doesn't have to be this PR

File: config/wiki.yaml [hunk 7]
```diff
@@ -18,13 +18,19 @@ ingress:
...
     memory: 128Mi
-replicaCount: 2
+
 image:
   pullPolicy: IfNotPresent
+
+replicaCount: 2
+
+nodeSelector:
+  agentpool: arm64small
```
[the remark this answers is not in the document]
Comment (smerle33) [reply config/wiki.yaml]: yes just javadoc

**Review by dduportal — APPROVED** [review]
LGTM 👍

**Pull request merged**